### PR TITLE
NS 7.0 with Angular 10 support

### PR DESCRIPTION
**Pull request by shansb-dev**
Hi madmas, 

I created a PR for NativeScript 7.0 which contains updated packages, further flattened imports and an additional build targeting Angular 10 applications.

To test the local demo application:
```
git clone https://github.com/shansb-dev/nativescript-pdf-view.git
cd nativescript-pdf-view
git checkout ns-7.0
cd src
npm run clean-setup
cd demo
npm install
ns run ios
ns run android
```

To test it with an angular sampleapp make sure to first create an Angular 10 compatible bundeled version.
From nativescript-pdf-view/src:
`npm run ng-packagr`

Create the sampleapp outside the nativescript-pdf-view project:
`ns create sampleapp --ng`

Include in tsconfig.json "include" property:
`"../nativescript-pdf-view/src"`

Add dependency in package.json:
`"nativescript-pdf-view": "file:../nativescript-pdf-view/src",`

Integrate PdfView call in sampleapp:
Add pdfUrl and onLoad() to item-detail.component.ts
```
public pdfUrl = 'https://www.princexml.com/howcome/2016/samples/magic8/index.pdf';

onLoad() {
   alert('Loaded PDF!');
}
```

Add PdfView in item-detail.component.html:
`<PDFView flexgrow="1" [src]="pdfUrl" (load)="onLoad()"></PDFView>
`

Add PDFViewModule to imports in app.module.ts:
```
import { PDFViewModule } from "nativescript-pdf-view/angular/dist";

@NgModule({
    ...
    imports: [
        ...
        PDFViewModule
    ],
    ...
})
```

Then install and run from sampleapp/src/app:
```
npm install
ns run ios
ns run android
```

When running it on an ios simulator on my machine I had an issue with @nativescript/ios@7.0.1 resulting in:
```
warning: There are no architectures to compile for because all architectures in VALID_ARCHS (i386, x86_64) are also in EXCLUDED_ARCHS (i386, armv6, armv8, x86_64, arm64e). (in target 'sampleapp' from project 'sampleapp')
** BUILD FAILED **
```
This can be solved by opening sampleapp/platforms/ios/sampleapp/xcworkspace go to User-Defined Build Settings and clearing the excluded archs field.

Please let me know if further changes are required.

Edit: Would fix Issue #23

### CODE REVIEW

**Comment by madmas** [comment]
Awesome, thanks a lot! I'll check this out ASAP, probably tonight or tomorrow.

**Comment by relez** [comment]
Great news, right now my apps are updated to NS 7 and my clients can't see pdf files in the app, hopefully they will soon. Thanks!

**Review by madmas — COMMENTED** [review]
Thank you for the PR, I think some minor tweaks are necessary. Can you add them to the PR? 

For Android, I get this exception in the sample app:

```
Installing on device 98UAY174QX...
Successfully installed on device with identifier '98UAY174QX'.
Restarting application on device 98UAY174QX...
JS: HMR: Hot Module Replacement Enabled. Waiting for signal.
Successfully synced application de.mynethome.nativescriptpdfviewdemo on device 98UAY174QX.
System.err: An uncaught Exception occurred on "main" thread.
System.err: Unable to start activity ComponentInfo{de.mynethome.nativescriptpdfviewdemo/com.tns.NativeScriptActivity}: com.tns.NativeScriptException: Calling js method onCreate failed
System.err: Error: Building UI from XML. @main-page.xml:7:5
System.err:  > Module 'nativescript-pdf-view' not found for element 'nativescript-pdf-view:PDFView'.
System.err:    > Cannot read property 'barteksc' of undefined
System.err: 
System.err: StackTrace:
System.err: java.lang.RuntimeException: Unable to start activity ComponentInfo{de.mynethome.nativescriptpdfviewdemo/com.tns.NativeScriptActivity}: com.tns.NativeScriptException: Calling js method onCreate failed
System.err: Error: Building UI from XML. @main-page.xml:7:5
System.err:  > Module 'nativescript-pdf-view' not found for element 'nativescript-pdf-view:PDFView'.
System.err:    > Cannot read property 'barteksc' of undefined
System.err:     at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3270)
System.err:     at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3409)
System.err:     at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:83)
System.err:     at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
System.err:     at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
System.err:     at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2016)
System.err:     at android.os.Handler.dispatchMessage(Handler.java:107)
System.err:     at android.os.Looper.loop(Looper.java:214)
System.err:     at android.app.ActivityThread.main(ActivityThread.java:7356)
System.err:     at java.lang.reflect.Method.invoke(Native Method)
System.err:     at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
System.err:     at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930)
System.err: Caused by: com.tns.NativeScriptException: Calling js method onCreate failed
System.err: Error: Building UI from XML. @main-page.xml:7:5
System.err:  > Module 'nativescript-pdf-view' not found for element 'nativescript-pdf-view:PDFView'.
System.err:    > Cannot read property 'barteksc' of undefined
System.err:     at com.tns.Runtime.callJSMethodNative(Native Method)
System.err:     at com.tns.Runtime.dispatchCallJSMethodNative(Runtime.java:1302)
System.err:     at com.tns.Runtime.callJSMethodImpl(Runtime.java:1188)
System.err:     at com.tns.Runtime.callJSMethod(Runtime.java:1175)
System.err:     at com.tns.Runtime.callJSMethod(Runtime.java:1153)
System.err:     at com.tns.Runtime.callJSMethod(Runtime.java:1149)
System.err:     at com.tns.NativeScriptActivity.onCreate(NativeScriptActivity.java:29)
System.err:     at android.app.Activity.performCreate(Activity.java:7825)
System.err:     at android.app.Activity.performCreate(Activity.java:7814)
System.err:     at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1306)
System.err:     at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3245)
System.err:     ... 11 more
```

Did it work for you?

**Comment by madmas** [comment]
About the Angular sample, there is https://github.com/madmas/nativescript-pdf-view-angular-sample - I^ll test it with that and upgrade that, too.

**Comment by relez** [comment]
Hi there, is there any update on this issue?

File: src/package.json [hunk 1]
```diff
@@ -6,8 +6,8 @@
   "typings": "index.d.ts",
   "nativescript": {
     "platforms": {
-      "android": "3.0.0",
-      "ios": "3.0.0"
+      "ios": "6.0.0",
```
[the remark this answers is not in the document]
Comment (shansb-dev) [reply src/package.json:9]: These platforms are added automatically when executing `ns platform add xy` and represent the lowest supported versions if I understand it correctely.

File: package-lock.json [hunk 2]
```diff
@@ -1,2476 +0,0 @@
-{
```
[the remark this answers is not in the document]
Comment (shansb-dev) [reply package-lock.json:1]: You are right this should be checked in.

File: src/angular/dist/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import { NativeScriptCommonModule } from "@nativescript/angular";
```
[the remark this answers is not in the document]
Comment (shansb-dev) [reply src/angular/dist/index.d.ts:1]: I think this should be checked in and probably also the compiled bundles.
If I am not mistaken the plugins that are delivered via npm are already compiled, therefore these files should be included.

File: src/__ivy_ngcc__/pdf-view.common.d.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import { Folder, Property, View } from '@nativescript/core';
```
[the remark this answers is not in the document]
Comment (shansb-dev) [reply src/__ivy_ngcc__/pdf-view.common.d.ts:1]: Probably for the same reason as src/angular/dist/... but I am not 100% sure

File: src/angular/dist/index.d.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (madmas) [reply src/angular/dist/index.d.ts:1]: There will be a build that produces the package that is distributed via NPM and as the GitHub release, so there is no need to have those generated file in the repository :-)

File: src/__ivy_ngcc__/pdf-view.common.d.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (madmas) [reply src/__ivy_ngcc__/pdf-view.common.d.ts:1]: see above

**Comment by madmas** [comment]
@shansb-dev did you try it with Android, do you get the same error as I do there?

**Comment by shansb-dev** [comment]
Yes I tried it with Android, the reasons was the `ns clean` command removed the gradle.include file in platforms folder that was responsible for this dependency. It works for me now.

**Comment by DanielAlcaraz** [comment]
Are there any intentions to continue with this?

**Comment by shansb-dev** [comment]
I think the other NS 7 PR (https://github.com/madmas/nativescript-pdf-view/pull/28) is more futureproof because it is build with the new recommended plugin workspace structure and was also already successfully adopted by a few users in the corresponding NS 7 issue. Im closing this for now.

**Pull request closed without merging**